src/sharechain/p2chain_level.rs

```diff
@@ -66,8 +66,8 @@ impl P2ChainLevel {
         Ok(())
     }
 
-    pub fn block_in_main_chain(&self) -> Option<&Arc<P2Block>> {
-        self.blocks.get(&self.chain_block)
+    pub fn block_in_main_chain(&self) -> Option<Arc<P2Block>> {
```

Comment:
I'm not sure I like this. Perhaps `get_block_in_main_chain`

Reply:
renamed